loopout.go

```diff
@@ -254,9 +257,25 @@ func (s *loopOutSwap) sendUpdate(ctx context.Context) error {
 func (s *loopOutSwap) execute(mainCtx context.Context,
 	cfg *executeConfig, height int32) error {
 
+	defer s.wg.Wait()
+
 	s.executeConfig = *cfg
 	s.height = height
 
+	// Create context for our state subscription which we will cancel once
+	// swap execution has completed, ensuring that we kill the subscribe
+	// goroutine.
+	subCtx, cancel := context.WithCancel(mainCtx)
+	defer cancel()
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		subscribeAndLogUpdates(
```

Comment:
Inside `subscribeAndLogUpdates`, it seems that another goroutine is started. Isn't one goroutine enough?

Reply:
Discussed offline that two goroutines is indeed ok. It is in line with how we build other grpc wrappers that convert `Recv` into channel sends